### PR TITLE
Snowbridge - Ethereum Client - Reject finalized updates without a sync committee in next store period

### PR DESCRIPTION
**Pull request by claravanstaden**
While syncing Ethereum consensus updates to the Snowbridge Ethereum light client, the syncing process stalled due to error `InvalidSyncCommitteeUpdate` when importing the next sync committee for period `1087`.

This bug manifested specifically because our light client checkpoint is a few weeks old (submitted to governance weeks ago) and had to catchup until a recent block. Since then, we have done thorough testing of the catchup sync process.

### Symptoms
- Import next sync committee for period `1086` (essentially period `1087`). Light client store period = `1086`.
- Import header in period `1087`. Light client store period = `1087`. The current and next sync committee is not updated, and is now in an outdated state. (current sync committee = `1086` and current sync committee = `1087`, where it should be current sync committee = `1087` and current sync committee = `None`)
- Import next sync committee for period `1087` (essentially period `1088`) fails because the expected next sync committee's roots don't match.

### Bug
The bug here is that the current and next sync committee's didn't handover when an update in the next period was received.

### Fix
There are two possible fixes here:
1. Correctly handover sync committees when a header in the next period is received.
2. Reject updates in the next period until the next sync committee period is known.

We opted for solution 2, which is more conservative and requires less changes.

### Polkadot-sdk versions
This fix should be backported in polkadot-sdk versions 1.7 and up.

Snowfork PR: https://github.com/Snowfork/polkadot-sdk/pull/145

### CODE REVIEW

**Comment by claravanstaden** [comment]
@vgeddes @yrong @alistair-singh please review.

**Review by vgeddes — APPROVED** [review]

File: bridges/snowbridge/pallets/ethereum-client/src/tests.rs [hunk 1]
```diff
@@ -362,23 +362,33 @@ fn submit_update_with_sync_committee_in_current_period() {
...
+			Error::<Test>::SyncCommitteeUpdateRequired
+		);
+		// submit update with next sync committee
+		assert_ok!(EthereumBeaconClient::submit(
+			RuntimeOrigin::signed(1),
+			next_sync_committee_update
+		));
+		// check same header in the next period can now be submitted successfully
 		assert_ok!(EthereumBeaconClient::submit(RuntimeOrigin::signed(1), update.clone()));
-		let block_root: H256 = update.finalized_header.clone().hash_tree_root().unwrap();
```
Comment (svyatonik) [inline bridges/snowbridge/pallets/ethereum-client/src/tests.rs:380]: Is it an intended removal? It looks like it should be fine even after the change?

**Review by svyatonik — APPROVED** [review]
The change itself is pretty clear and the issue is very similar to what we have in K<>P bridge - we've accepted the header without updating the authorities set (sync committee).

File: bridges/snowbridge/pallets/ethereum-client/src/tests.rs (hunk 1 above)
Comment (claravanstaden) [reply bridges/snowbridge/pallets/ethereum-client/src/tests.rs:380]: Good point! Added back in https://github.com/paritytech/polkadot-sdk/pull/4478/commits/9fdcef31dc12f42e6499e460899ab67b21961231.

**Review by acatangiu — APPROVED** [review]

**Review by bkontur — APPROVED** [review]
when merged, we should backport to all other:
- [ ] release-crates-io-v1.7.0 - patch release the fellows BridgeHubs runtimes - @claravanstaden do you want to do it?
- [ ] release-crates-io-v1.8.0
- [ ] release-crates-io-v1.9.0
- [ ] release-crates-io-v1.10.0
- [ ] release-crates-io-v1.11.0
- [ ] release-crates-io-v1.12.0 (commit soon)

**Comment by claravanstaden** [comment]
> when merged, we should backport to the all other release branches:
> 
> * [ ]  release-crates-io-v1.7.0 - patch release the fellows BridgeHubs runtimes - @claravanstaden do you want to do it?
> * [ ]  release-crates-io-v1.8.0
> * [ ]  release-crates-io-v1.9.0
> * [ ]  release-crates-io-v1.10.0
> * [ ]  release-crates-io-v1.11.0
> * [ ]  release-crates-io-v1.12.0 (commit soon)

@bkontur I'll all of the above, yes. I need to update the Ethereum client storage in the runtimes PR too.

**Comment by claravanstaden** [comment]
@bkontur rest of the backported fix PRs:

- 1.8.0 - https://github.com/paritytech/polkadot-sdk/pull/4483
- 1.9.0 - https://github.com/paritytech/polkadot-sdk/pull/4484
- 1.10.0 - https://github.com/paritytech/polkadot-sdk/pull/4485
- 1.11.0 - https://github.com/paritytech/polkadot-sdk/pull/4486

**Pull request merged**